dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+
+    """
+
+    # Config parameters related to scheduling
+    lag = config.Property(proptype=str2timedelta, default="4h")
+
+    # Config parameters defining data file selection
+    correlator = config.Property(proptype=str, default="chime")
+    acq_suffix = config.Property(proptype=str, default="stack_corr")
+    source_transits = config.Property(proptype=list, default=["cyg_a"])
+
+    # Config parameters defining output data product
+    output_suffix = config.Property(proptype=str, default="sensitivity")
+
+    # Config parameters related to cylinders
+    cyl_start_char = config.Property(proptype=int, default=65)
+    cyl_start_num = config.Property(proptype=int, default=2)
+    sep_cyl = config.Property(proptype=bool, default=False)
+    rot = config.Property(proptype=float, default=-0.088)
+
+    # Config parameters related to optional inclusions
+    include_auto = config.Property(proptype=bool, default=False)
+    include_intracyl = config.Property(proptype=bool, default=False)
+    include_crosspol = config.Property(proptype=bool, default=False)
+
+    # Config parameters related to the algorithm
+    nfreq_per_block = config.Property(proptype=int, default=16)
+    nsigma = config.Property(proptype=float, default=1.0)
+    perform_fit = config.Property(proptype=bool, default=False)
+
+    def setup(self):
+        """
+        Set up the task.
+
+        Creates metrics.
+        """
+        self.run_metric = self.add_task_metric(
+            "source_analyzer_run_total",
+            "task run counter for specific sources",
+            labelnames=["source"],
+            unit="total",
+        )
+
+        self.logger.info("Sources: {}".format(self.source_transits))
+
+    def run(self):
+        """Run the analyzer.
+
+        Load autocorrelations for a given source transit.
+        Extract flux of a given source, as a function of freq,
+        when the source is right at the zenith.
+
+        Write fluxes for a given time and frequency to disk.
+        """
+        lat = np.radians(ephemeris.CHIMELATITUDE)
+        err_msg = ""
+
+        # Create transit tracker
+        for src in self.source_transits:
+            try:
+                src_body = fluxcat.FluxCatalog[src].skyfield
+            except KeyError as e:
+                raise DiasConfigError("Invalid source: {}".format(e))
+
+            # initalise run metric
+            self.run_metric.labels(source=src).set(0)
+
+            self.logger.info(
+                "Initializing offline point source processing for %s.".format(src)
+            )
+
+            # Query files from now to period hours back
+            query_stop_time = datetime.utcnow() - self.lag
+            query_start_time = query_stop_time - self.period
+
+            self.logger.info(
+                "Searching for transits from %s to %s"
+                % (str(query_start_time), str(query_stop_time))
+            )
+
+            window = self.nsigma * cal_utils.guess_fwhm(
```

Comment:
The window can be returned in units of seconds with
```
time_delta = 2.0 * self.nsigma * cal_utils.guess_fwhm(
400.0, pol="X", dec=src_body.dec.radians, sigma=True, seconds=True)
```
since you never use the value in degrees.